file/src/main/scala/akka/stream/alpakka/file/impl/archive/TarReaderStage.scala

```diff
@@ -37,12 +36,15 @@ private[file] class TarReaderStage
           extends SubSourceOutlet[ByteString]("fileOut")
           with TarReaderStage.SourceWithTimeout
 
-      readHeader(ByteString.empty)
+      setHandlers(flowIn, flowOut, new CollectHeader(ByteString.empty))
```

Comment:
We can't do a pull in the constructor, only set handlers, so instead of calling `readHeader` we set collect handlers here directly, we anyway know the buffer is empty.